src_c/draw.c

```diff
@@ -1020,53 +1020,58 @@ set_pixel_32(Uint8 *pixels, SDL_PixelFormat *format, Uint32 pixel)
 }
 
 static void
-draw_pixel_blended_32(Uint8 *pixels, Uint8 *colors, float br,
-                      SDL_PixelFormat *format)
+draw_shaded_32_pixel(SDL_Surface *surf, Uint8 *pixels, int x, int y, int blend,
+                     Uint8 *colors, float br)
 {
-    Uint8 pixel32[4];
-    SDL_GetRGBA(get_pixel_32(pixels, format), format, &pixel32[0], &pixel32[1],
-                &pixel32[2], &pixel32[3]);
-    *(Uint32 *)pixel32 =
-        SDL_MapRGBA(format, br * colors[0] + (1 - br) * pixel32[0],
-                    br * colors[1] + (1 - br) * pixel32[1],
-                    br * colors[2] + (1 - br) * pixel32[2],
-                    br * colors[3] + (1 - br) * pixel32[3]);
-    set_pixel_32(pixels, format, *(Uint32 *)pixel32);
-}
 
-#define DRAWPIX32(pixels, colorptr, br, blend)                              \
-    {                                                                       \
-        if (blend)                                                          \
-            draw_pixel_blended_32(pixels, colorptr, br, surf->format);      \
-        else {                                                              \
-            set_pixel_32(                                                   \
-                pixels, surf->format,                                       \
-                SDL_MapRGBA(surf->format, br *colorptr[0], br *colorptr[1], \
-                            br *colorptr[2], br *colorptr[3]));             \
-        }                                                                   \
+    if (x < surf->clip_rect.x || x >= surf->clip_rect.x + surf->clip_rect.w ||
+        y < surf->clip_rect.y || y >= surf->clip_rect.y + surf->clip_rect.h) {
+        // the (x, y) can happen to be a bit over the border because we need
+        // the extra space to make draw_aaline work correctly on the border
+        return;
     }
-
+    SDL_PixelFormat* format = surf->format;
```

Comment:
Some old C compilers require that variables be declared at the start of a bracket/block, so at the start of the function in this case.
That's probably why the first compilation failed. I don't know why the other tests failed.

Reply:
Until now, no test failed because no test was executed :) I fixed the variable declaration. Let's see if it changes something.

Reply:
Well, it didn't change anything...
Unfortunately, we can not do some *bisect* because all the C code is in one commit, and it's a complete re-write. It must _somehow_ have to do with the automated-testing framsworks because on my Debian **stretch** it all works and all tests pass!

```bash
$ uname -a
Linux terbium 4.9.0-8-amd64 #1 SMP Debian 4.9.130-2 (2018-10-27) x86_64 GNU/Linux

$ python
Python 3.5.3 (default, Jan 19 2017, 14:11:04) 
[GCC 6.3.0 20170118] on linux
```